tests/test_pipeline_mixin.py

```diff
@@ -236,67 +311,89 @@ def run_pipeline_test(
+
         if TRANSFORMERS_TINY_MODEL_PATH != "hf-internal-testing":
-            model_type = model_architecture.config_class.model_type
             repo_id = os.path.join(TRANSFORMERS_TINY_MODEL_PATH, model_type, repo_name)
 
+        # -------------------- Load model --------------------
+
+        # TODO: We should check if a model file is on the Hub repo. instead.
+        try:
+            model = model_architecture.from_pretrained(repo_id, revision=commit)
+        except Exception:
+            logger.warning(
+                f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: Could not find or load "
+                f"the model from `{repo_id}` with `{model_architecture}`."
+            )
+            self.skipTest(f"Could not find or load the model from {repo_id} with {model_architecture}.")
+
+        # -------------------- Load tokenizer --------------------
+
         tokenizer = None
         if tokenizer_name is not None:
-            tokenizer_class = getattr(transformers_module, tokenizer_name)
-            tokenizer = tokenizer_class.from_pretrained(repo_id, revision=commit)
-
-        processor = None
-        if processor_name is not None:
-            processor_class = getattr(transformers_module, processor_name)
-            # If the required packages (like `Pillow` or `torchaudio`) are not installed, this will fail.
             try:
-                processor = processor_class.from_pretrained(repo_id, revision=commit)
+                tokenizer_class = getattr(transformers_module, tokenizer_name)
+                tokenizer = tokenizer_class.from_pretrained(repo_id, revision=commit)
             except Exception:
                 logger.warning(
-                    f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: Could not load the "
-                    f"processor from `{repo_id}` with `{processor_name}`."
+                    f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: "
+                    f"Could not load the tokenizer from `{repo_id}` with `{tokenizer_name}`."
                 )
-                self.skipTest(f"Could not load the processor from {repo_id} with {processor_name}.")
+                self.skipTest(f"Could not load the tokenizer from {repo_id} with {tokenizer_name}.")
+
+        # -------------------- Load processors --------------------
+
+        processors = {}
+        for key, name in zip(
+            ["image_processor", "feature_extractor", "processor"],
+            [image_processor_name, feature_extractor_name, processor_name],
+        ):
+            if name is not None:
+                try:
+                    # Can fail if some extra dependencies are not installed
+                    processor_class = getattr(transformers_module, name)
+                    processor = processor_class.from_pretrained(repo_id, revision=commit)
```

Comment:
we should make sure `processor` is an instance of `ProcessorMixin`, otherwise reset it to `None`

Reply:
should we? Class is not checked for feature extractor or image processor, I guess we might want to load anything that might be loaded with AutoProcessor class while it keeps the pipeline working.

Reply:
> Class is not checked for feature extractor or image processor

This is because they are not a composite, and are not design to fallback to load partially.

For example, this works
```
from transformers import AutoProcessor
AutoProcessor.from_pretrained("bert-base-uncased")
```
but this fails
```
from transformers import AutoImageProcessor
AutoImageProcessor.from_pretrained("bert-base-uncased")
```
and same
```
from transformers import AutoTokenizer
AutoTokenizer.from_pretrained("google/vit-base-patch16-224")
```

Reply:
Got it, addressed in https://github.com/huggingface/transformers/pull/32514/commits/2f561479290970fa8beaf668b504df39a2608f4f